src/Webtoon/WebtoonParser.ts

```diff
@@ -102,34 +94,27 @@ export abstract class WebtoonParser extends WebtoonSettings {
         return $("#content > div.cont_box span.thmb > img").attr("src") ?? "";
     }
 
-    parseChaptersList($: CheerioAPI, sourceManga: SourceManga): Chapter[] {
-        return $("ul#_episodeList > li[id*=episode]")
-            .toArray()
-            .map((elem) => this.parseChapter($(elem), sourceManga));
+    parseChaptersList(
+        dto: WebtoonChaptersListDto,
+        sourceManga: SourceManga,
+    ): Chapter[] {
+        return dto.episodeList.map((elem) =>
+            this.parseChapter(elem, sourceManga),
+        );
     }
 
-    parseChapter(elem: CheerioElement, sourceManga: SourceManga): Chapter {
+    parseChapter(
+        elem: WebtoonChaptersElemDto,
+        sourceManga: SourceManga,
+    ): Chapter {
         return {
-            chapterId:
-                elem
-                    .find("a")
-                    .attr("href")
-                    ?.replace(this.MOBILE_URL + "/", "") ?? "",
+            chapterId: elem.viewerLink,
```

Comment:
One last thing I just saw: We need a elem.viewerLink.replace("/", "") here, so the chapter id doesn't start with a slash all of a sudden. That will otherwise cause the loss of reading progress